src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health, hidden);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const targetAnimal = Animal.alive.find((animal) => animal === target);
+    const targetAnimalIndex = Animal.alive.indexOf(targetAnimal);
+
+    if (targetAnimal.hidden === true || targetAnimal instanceof Carnivore) {
+      return;
+    }
```

Comment:
Try to rewrite condition the way that most of your code would be in `if` statement. You don't need simply `return`